nengo/cache.py

```diff
@@ -133,11 +133,12 @@ def sync(self):
                 for key in self._deletes:
                     del self._index[key]
 
-                with open(self.filename, 'wb') as f:
+                with open(self.filename + '.part', 'wb') as f:
                     pickle.dump(
                         {k: v for k, v in self._index.items()
                          if v[0] not in self._removed_files},
                         f, pickle.HIGHEST_PROTOCOL)
+                os.rename(self.filename + '.part', self.filename)
```

Comment:
Is `self.filename` guaranteed not to exist? That's the only flaw I can see in this implementation. In Unix, the rename will overwrite, but on Windows it won't ([reference](http://stackoverflow.com/questions/8107352/force-overwrite-in-os-rename)).